### PR TITLE
minor fix about scrubEphemeralDisk

### PR DESCRIPTION
**Pull request by zhanggbj**
1. move scrubEphemeralDisk after mounting disk
2. scrub ephemeral disk  when to upgrade from old stemcell without
   ScrubEphemeralDisk enabled.

### CODE REVIEW

**Comment by zhanggbj** [comment]
Hi Dmitriy @cppforlife,

We are in the progress of upgrading Bluemix and it depends on this PR, would you please take a review on it when you got time? Thank you very much:)

cc @mattcui

**Comment by cppforlife** [comment]
@zhanggbj will try to get this merged in shortly.

File: platform/linux_platform.go [hunk 1]
```diff
@@ -600,6 +593,18 @@ func (p linux) SetupEphemeralDiskWithPath(realPath string) error {
 		return bosherr.WrapError(err, "Mounting data partition")
 	}
 
+	if p.options.ScrubEphemeralDisk {
+		contents, err := p.fs.Glob(mountPointGlob)
+		if err != nil {
+			return bosherr.WrapErrorf(err, "Globbing ephemeral disk mount point `%s'", mountPointGlob)
```
Comment (ronakbanka) [inline platform/linux_platform.go]: @zhanggbj Just curious,  what is with this `%s' instead of '%s' ??

Comment (zhanggbj) [inline platform/linux_platform.go]: sorry, this is a typo. fixed it.

**Comment by zhanggbj** [comment]
Hi Dmitriy @cppforlife,

Sorry for pushing, but we are waiting for this PR merged to continue upgrading Bluemix. Would you please take a look? Thanks for your help:)

cc @mattcui

**Comment by tylerschultz** [comment]
Hi @zhanggbj, would you please update the unit tests for this change? Thank you.

File: platform/linux_platform_test.go [hunk 2]
```diff
@@ -1066,6 +1075,26 @@ fake-base-path/data/sys/log/*.log fake-base-path/data/sys/log/.*.log fake-base-p
 				Context("when agent_version file does not exist", func() {
 					BeforeEach(func() {
 						fs.RemoveAll(path.Join(dirProvider.DataDir(), ".bosh", "agent_version"))
+						fs.SetGlob(path.Join("/fake-dir", "data", "", "*"), []string{"/fake-dir/data/fake-file1", "/fake-dir/data/fakedir"})
```
Comment (dpb587-pivotal) [inline platform/linux_platform_test.go]: Is the empty file name between `data` and `*` necessary?

Comment (zhanggbj) [inline platform/linux_platform_test.go]: Removed it. Thanks.

**Comment by zhanggbj** [comment]
Hi @cppforlife @ronakbanka @tylerschultz @dpb587-pivotal ,

All comments are addressed and would you please help to take a another review? Thank you.

/cc @maximilien

**Comment by cppforlife** [comment]
Story: https://www.pivotaltracker.com/story/show/134285049

**Comment by tjvman** [comment]
@zhanggbj Thanks for the submission! We've merged in this PR against the `develop` branch.

In the future, please make pull requests against the develop branch - we do not merge PRs against master.

**Pull request closed without merging**